### PR TITLE
agent,plugin: Don't send ComputeUnit from plugin

### PR DESCRIPTION
**Pull request by sharnoff**
Instead, define it in the autoscaler-agent config.

This change starts the removal process of ComputeUnit from PluginResponse, so that the new protocol version expects that it won't be set - but to preserve backwards-compatibility during the upgrade, we need to keep the old behavior for now (hence the field is `omitempty`).

A follow-up change will update the plugin so that it *never* sends ComputeUnit, removing the field and the plugin's config along with it.

Part of #706.

---

Notes for review: This overlaps _a little_ with #653 — I think it's _probably_ easier to merge this first, release, and then merge that one.

~~Marking as a draft for now, until I've tested that e2e tests pass with old agent + new scheduler.~~ Tested locally and it appears to work.

### CODE REVIEW

File: pkg/api/types.go [hunk 1]
```diff
@@ -58,9 +58,18 @@ const (
 	//
 	// * added AgentRequest.LastPermit
 	//
-	// Currently the latest version.
+	// Last used in release version <todo>.
```
Comment (sharnoff) [inline pkg/api/types.go:61]: > `<todo>`

Leaving this for now, will set it before merging

File: pkg/agent/core/state_test.go [hunk 2]
```diff
@@ -1418,7 +1415,23 @@ func TestMetricsConcurrentUpdatedDuringDownscale(t *testing.T) {
 	}
 
 	// Send initial scheduler request - without the monitor active, so we're stuck at 4 CU for now
-	doInitialPluginRequest(a, state, clock, duration("0.1s"), DefaultComputeUnit, nil, resForCU(4))
+	a.
```
Comment (shayanh) [inline pkg/agent/core/state_test.go:1418]: I guess this test is not directly related to this PR but it's fine.

**Review by shayanh — APPROVED** [review]
lgtm

File: pkg/api/types.go (hunk 1 above)
Comment (Omrigan) [reply pkg/api/types.go:61]: And update `pkg/api/VERSIONING.md` probably

File: pkg/agent/core/state_test.go (hunk 2 above)
Comment (sharnoff) [reply pkg/agent/core/state_test.go:1418]: Ah, right - I did this because the expected warnings changed here. But looking at this again, just realized that I can actually just pass `a.WithWarnings(...)` into `doInitialPluginRequest`, which should make this simpler.

File: pkg/api/types.go (hunk 1 above)
Comment (sharnoff) [reply pkg/api/types.go:61]: Ah, yep! good catch - thanks!
